Ability to use custom API endpoint

jlestel opened the pull request:
As already discussed #255 #292 the official Tesla API will be the only way to send commands.

Using third-party Tesla API providers (or a self-hosted commands proxy) would allow command endpoints to be preserved.

I added an environment variable to allow users to choose their API provider.

tobiasehlert commented:
hi @jlestel,

Thanks for your pull request!

Just of curiosity, is this to be able to send the requests through MyTeslamate towards the Fleet API endpoint or what is the goal?
I'm considering either removing the commands section from TeslaMateAPI or yeah.. try to change it so that it can be used with the Fleet API directly (even though I don't use the commands endpoints anymore).

Kind regards,
Tobias

tobiasehlert commented on README.md:
```diff
@@ -122,6 +122,7 @@ Basically the same environment variables for the database, mqqt and timezone nee
 | **MQTT_PASSWORD**     | string  |                               |
 | **MQTT_NAMESPACE**    | string  |                               |
 | **MQTT_CLIENTID**     | string  | _4 char random string_        |
+| **TESLA_API_URL**     | string  | _uses Owner API by default_   |
```
```suggestion
| **TESLA_API_HOST**    | string  | _uses Owner API by default_   |
```

jlestel replied:
updated

tobiasehlert commented on src/v1_TeslaMateAPICarsCommand.go:
```diff
@@ -112,9 +112,9 @@ func TeslaMateAPICarsCommandV1(c *gin.Context) {
 	switch getCarRegionAPI(TeslaAccessToken) {
 	case ChinaAPI:
-		TeslaEndpointUrl = "https://owner-api.vn.cloud.tesla.cn"
+		TeslaEndpointUrl = getEnv("TESLA_API_URL", "https://owner-api.vn.cloud.tesla.cn")
```
```suggestion
		TeslaEndpointUrl = getEnv("TESLA_API_HOST", "https://owner-api.vn.cloud.tesla.cn")
```

tobiasehlert commented on src/v1_TeslaMateAPICarsCommand.go:
```diff
@@ -112,9 +112,9 @@ func TeslaMateAPICarsCommandV1(c *gin.Context) {
+		TeslaEndpointUrl = getEnv("TESLA_API_URL", "https://owner-api.vn.cloud.tesla.cn")
 	default:
-		TeslaEndpointUrl = "https://owner-api.teslamotors.com"
+		TeslaEndpointUrl = getEnv("TESLA_API_URL", "https://owner-api.teslamotors.com")
```
```suggestion
		TeslaEndpointUrl = getEnv("TESLA_API_HOST", "https://owner-api.teslamotors.com")
```

tobiasehlert reviewed (commented):
Two things regarding this pull request, what do you think?

1. I prefer use the same environment variable names as used in the TeslaMate project for reusability within the setup (pushed in 5b32d3e).
2. Would be good to have some info-message at startup that says that the `TESLA_API_HOST` environment variable is set and therefore is overriding the API command calls.

Other than that it looks good to me :)

jlestel commented:
> hi @jlestel,
> 
> Thanks for your pull request!
> 
> Just of curiosity, is this to be able to send the requests through MyTeslamate towards the Fleet API endpoint or what is the goal? I'm considering either removing the commands section from TeslaMateAPI or yeah.. try to change it so that it can be used with the Fleet API directly (even though I don't use the commands endpoints anymore).
> 
> Kind regards, Tobias

no I already send commands via myteslamate. the goal is rather to propose an oss mobile application for teslamate via teslamateapi.

so if it allows users who have their api command to use it to send commands with teslamateapi.

tobiasehlert commented:
@jlestel, are you adding the suggestions? I thought about releasing a new version later today/tomorrow :)

jlestel commented:
Hi Tobias, I try to improve it tonight

jlestel commented:
I changed the env var name. And only if it's set, an info log line display the value of TESLA_API_HOST on launch.

I fixed a typo on the previous line too: https://github.com/tobiasehlert/teslamateapi/pull/307/files#r1845637370
